sqs/sqs.py

```diff
@@ -30,43 +30,56 @@ def name(self):
     def url(self):
         return self._queue_url
 
-    def _get_queue(self, queue_name):
+    def __get_queue(self, queue_name):
         """ Returns a queue from ``region`` with ``queue_name`` """
         return self._resource.get_queue_by_name(QueueName=queue_name)
 
     @staticmethod
-    def is_empty(queue_attrs):
-        """Checks for the ``ApproxmiateNumberOfMessages`` in the queue attributes
-        and returns True when the value is 0 or if the attribute is not found.
+    def __is_empty(queue_attrs):
+        """Returns true if the queue is determined to be empty, otherwise
+        False. To determine whether or not the queue is empty, we look at the
+        attributes named ``ApproximateNumberOfMessages`` and
+        ``ApproxmiateNumberOfMessagesNotVisible``.
         """
         approx = sum(int(queue_attrs.get(m, '0')) for m in [
             'ApproximateNumberOfMessages',
             'ApproximateNumberOfMessagesNotVisible'])
         return True if approx == 0 else False
 
-    def inspect_queue(self, attribute_names=None):
-        """ Inspects a queue and returns whether or not the queue is empty """
+    def __get_attributes(self, attribute_names=None):
+        """ Dynamically fetches new attributes for the queue. If
+        ``attribute_names`` is not provided, we fetch all attributes.
+        """
         if not attribute_names:
             attribute_names = ['All']
 
+        if not isinstance(attribute_names, list):
+            attribute_names = [attribute_names]
+
         client = self._queue.meta.client
 
-        return SQSClient.is_empty(
-                client.get_queue_attributes(
+        return client.get_queue_attributes(
                     QueueUrl=self.url,
                     AttributeNames=attribute_names
                     ).get('Attributes')
-                )
 
-    def purge(self):
+    def purge_queue(self):
         """ Clears a queue ``queue_name`` of ALL messages. This action is
         not reversable """
         self._queue.purge()
 
-    def queue_should_be(self, empty):
-        """ Asserts that the queue is either empty or not """
-        expected = True if empty == 'empty' else False
-        actual = self.inspect_queue()
-        if actual != expected:
-            raise AssertionError("Expected the queue to be {} but, got {}".format(
-                empty, actual))
+    def queue_should_be_empty(self):
+        """ Asserts that the queue is empty """
+        if not SQSClient.__is_empty(self.__get_attributes()):
+            raise AssertionError("The queue was not empty")
+
+    def queue_contains_visible_messages(self, messages_no=1):
```

Comment:
It is good practise to only raise exceptions in keywords that contain the word "should".
In this case I would consider having a keyword called "get number of visible messages" that always returns the count and then write "Should be true  ${count} > 2" in the test case itself.